flutter-for-android.md

```diff
@@ -0,0 +1,1845 @@
+
+  @override
+  _SampleAppPageState createState() => new _SampleAppPageState();
+}
+
+class _SampleAppPageState extends State<SampleAppPage> {
+  String _errorText;
+
+  @override
+  Widget build(BuildContext context) {
+    return new Scaffold(
+      appBar: new AppBar(
+        title: new Text("Sample App"),
+      ),
+      body: new Center(
+        child: new TextField(
+          onSubmitted: (String text) {
+            setState(() {
+              if (!isEmail(text)) {
+                _errorText = 'Error: This is not an email';
+              } else {
+                _errorText = null;
+              }
+            });
+          },
+          decoration: new InputDecoration(hintText: "This is a hint", errorText: _getErrorText()),
+        ),
+      ),
+    );
+  }
+
+  _getErrorText() {
+    return _errorText;
+  }
+
+  bool isEmail(String em) {
+    String p =
+        r'^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$';
+    
+    RegExp regExp = new RegExp(p);
+    
+    return regExp.hasMatch(em);
+  }
+}
+{% endprettify %}
+
+
+# Flutter Plugins
+
+**How do I access the GPS sensor?**
+
+Flutter does not access the GPS sensor directly, instead it relies on Plugins. 
+
+To access the GPS sensor you can build your own plugin [Flutter Plugin how to] 
+or use a community plugin https://pub.dartlang.org/packages/location
+
+**How do I access the Camera?**
+
+A popular community plugin to access the camera is 
+https://pub.dartlang.org/packages/image_picker
+
+**How do I log in with Facebook**
+
+To access Facebook Connect functionality you can use 
+https://pub.dartlang.org/packages/flutter_facebook_connect
+
+**How do I build my own custom native integrations?**
+
+If there is platform specific functionality that Flutter or its community 
+Plugins are missing then you can build your own following this tutorial 
+https://flutter.io/developing-packages/ .
+
+Flutters plugin architecture in a nutshell is a lot like using an Event bus in 
+Android, you fire off a message and let the receiver process and emit a result 
+back to you, in this case the receiver would be iOS or Android.
+
+**How do I use the NDK in my Flutter application**
+
+If you are using the NDK in your current application and want to take advatnage 
+of your native libraries in Flutter, then you can build a custom plugin that 
+hooks into Android, which then talks to the NDK, returns the result to Android, 
+which then propagates the result back up to Flutter.
+
+# Themes
+
+**How do I theme my app?**
+
+Flutter out of the box comes with a precise implementation of Material Design, 
+which takes care of a lot of styling and theming needs that you would typically 
+do. Unlike Android where you declare themes in XML and then assign it to your 
+application via AndroidManifest.xml, in Flutter you declare themes via the top 
```

Comment:
"you can declare"

using the top-level widget isn't the only way. so s/you declare/you can declare